phpBB/phpbb/template/twig/node/event.php

```diff
@@ -42,40 +46,80 @@ public function compile(\Twig\Compiler $compiler)
 
 		$location = $this->listener_directory . $this->getNode('expr')->getAttribute('name');
 
+		$compiler_steps = [];
+
 		foreach ($this->environment->get_phpbb_extensions() as $ext_namespace => $ext_path)
 		{
 			$ext_namespace = str_replace('/', '_', $ext_namespace);
 
+			$compiler_calls = [];
+
 			if ($this->environment->isDebug())
 			{
 				// If debug mode is enabled, lets check for new/removed EVENT
 				//  templates on page load rather than at compile. This is
 				//  slower, but makes developing extensions easier (no need to
 				//  purge the cache when a new event template file is added)
-				$compiler
-					->write("if (\$this->env->getLoader()->exists('@{$ext_namespace}/{$location}.html')) {\n")
-					->indent()
-				;
+				$compiler_calls[] = function() use($compiler, $ext_namespace, $location) {
+					$compiler
+						->write("if (\$this->env->getLoader()->exists('@{$ext_namespace}/{$location}.html')) {\n")
+						->indent()
+					;
+				};
 			}
 
 			if ($this->environment->isDebug() || $this->environment->getLoader()->exists('@' . $ext_namespace . '/' . $location . '.html'))
 			{
-				$compiler
-					->write("\$previous_look_up_order = \$this->env->getNamespaceLookUpOrder();\n")
-
-					// We set the namespace lookup order to be this extension first, then the main path
-					->write("\$this->env->setNamespaceLookUpOrder(array('{$ext_namespace}', '__main__'));\n")
-					->write("\$this->env->loadTemplate('@{$ext_namespace}/{$location}.html')->display(\$context);\n")
-					->write("\$this->env->setNamespaceLookUpOrder(\$previous_look_up_order);\n")
-				;
+				$compiler_calls[] = function() use($compiler, $ext_namespace, $location) {
+					$compiler
+						->write("\$previous_look_up_order = \$this->env->getNamespaceLookUpOrder();\n")
+
+						// We set the namespace lookup order to be this extension first, then the main path
+						->write("\$this->env->setNamespaceLookUpOrder(array('{$ext_namespace}', '__main__'));\n")
+						->write("\$this->env->loadTemplate('@{$ext_namespace}/{$location}.html')->display(\$context);\n")
+						->write("\$this->env->setNamespaceLookUpOrder(\$previous_look_up_order);\n")
+					;
+				};
 			}
 
 			if ($this->environment->isDebug())
 			{
-				$compiler
-					->outdent()
-					->write("}\n\n")
-				;
+				$compiler_calls[] = function() use($compiler) {
+					$compiler
+						->outdent()
+						->write("}\n\n")
+					;
+				};
+			}
+
+			if (!empty($compiler_calls))
+			{
+				if (isset($this->template_event_priority_array[$ext_namespace][$location]))
+				{
+					$priority_key = $this->template_event_priority_array[$ext_namespace][$location];
```

Comment:
Priority may not be set for the majority of template events, hence this assignment will cause `Undefined array key` PHP warnings. So it probably needs to be something like
```php
$priority_key = $this->template_event_priority_array[$ext_namespace][$location] ?? 0;
```

Reply:
That's fine as is, if it's not set, it goes to [the else on line 112](https://github.com/phpbb/phpbb/blob/31fb27e8d88a1ad51ce1dbdeb5845054e14262bb/phpBB/phpbb/template/twig/node/event.php#L112):

```php
if (isset($this->template_event_priority_array[$ext_namespace][$location]))
{
	...
}
else
{
	array_unshift($compiler_steps, $compiler_calls);
}
```

If it's unset, `array_unshift` places the current `$compiler_calls` at the front of the `$compiler_steps` array. This ensures that the highest number key always takes precedence. In the screenshot I included in [my first comment on this pr](https://github.com/phpbb/phpbb/pull/6571#issuecomment-1837608225), you can see that the other extension's `overall_header_head_append` are indeed executed in their original order, and they do not make use of this event.

Reply:
That's ok, but it'll PHP erroring on assignment itself.